src/content-helper/common/utils/date.ts

```diff
@@ -13,20 +13,14 @@ const DATE_NOT_AVAILABLE_MESSAGE = __( 'Date N/A', 'wp-parsely' );
  * @param {Date} date The date to be examined.
  * @return {boolean} Whether the date can be processed further.
  */
-export function canProcessDate( date: Date|string ): boolean {
-	if ( 'string' === typeof date ) {
-		// Verify that date format is "YYYY-MM-DD".
-		if ( ! /^\d{4}-\d{2}-\d{2}$/.test( date ) ) {
-			return false;
-		}
-
-		date = new Date( date );
+export function canProcessDate( date: Date ): boolean {
+	// Return false if the date is not a valid Date object.
+	if ( isNaN( +date ) ) {
```

Comment:
It seems like we can remove this `canProcessDate` function and use `isNaN( date )` directly to check if the data is valid or not.

Reply:
Having a named function makes code more semantic, and also allows us to add to or change our logic in the future more easily if needed.